pkg/controller/certificaterequests/sync_test.go

```diff
@@ -306,79 +306,91 @@ func TestSync(t *testing.T) {
+					gen.Issuer("ca-issuer",
+						gen.AddIssuerCondition(cmapi.IssuerCondition{
+							Type:   cmapi.IssuerConditionReady,
+							Status: cmapi.ConditionTrue,
+						}),
+						gen.SetIssuerCA(cmapi.CAIssuer{}),
+					),
+				},
 				ExpectedActions: []testpkg.Action{
 					testpkg.NewAction(coretesting.NewUpdateAction(
 						cmapi.SchemeGroupVersion.WithResource("certificaterequests"),
 						gen.DefaultTestNamespace,
 						exampleCRReadyConditionWithGroupRef,
-					)),
+					),
+					),
 				},
 			},
 			CheckFn: func(t *testing.T, s *controllerFixture, args ...interface{}) {
 			},
 			Err: false,
 		},
 		"should exit sync nil if issuerRef group does not match certmanager.k8s.io": {
-			Issuer: gen.Issuer("test",
-				gen.AddIssuerCondition(cmapi.IssuerCondition{
-					Type:   cmapi.IssuerConditionReady,
-					Status: cmapi.ConditionTrue,
-				}),
-				gen.SetIssuerSelfSigned(cmapi.SelfSignedIssuer{}),
-			),
-			CertificateRequest: *exampleCRWrongIssuerRefGroup,
+			CertificateRequest: exampleCRWrongIssuerRefGroup,
 			IssuerImpl: &fake.Issuer{
 				FakeSign: func(context.Context, *cmapi.CertificateRequest) (*issuer.IssueResponse, error) {
 					return nil, errors.New("unexpected sign call")
 				},
 			},
 			Builder: &testpkg.Builder{
-				CertManagerObjects: []runtime.Object{gen.CertificateRequest("test")},
-				ExpectedActions:    []testpkg.Action{}, // no update
+				CertManagerObjects: []runtime.Object{gen.CertificateRequest("test"),
+					gen.Issuer("ca-issuer",
+						gen.AddIssuerCondition(cmapi.IssuerCondition{
+							Type:   cmapi.IssuerConditionReady,
+							Status: cmapi.ConditionTrue,
+						}),
+						gen.SetIssuerCA(cmapi.CAIssuer{}),
+					),
+				},
+				ExpectedActions: []testpkg.Action{}, // no update
 			},
 			CheckFn: func(t *testing.T, s *controllerFixture, args ...interface{}) {
 			},
 			Err: false,
 		},
 		"should not update certificate request if certificate exists, even if out of date": {
-			Issuer: gen.Issuer("test",
-				gen.AddIssuerCondition(cmapi.IssuerCondition{
-					Type:   cmapi.IssuerConditionReady,
-					Status: cmapi.ConditionTrue,
-				}),
-				gen.SetIssuerSelfSigned(cmapi.SelfSignedIssuer{}),
-			),
-			CertificateRequest: *exampleSignedExpiredCR,
+			CertificateRequest: exampleSignedExpiredCR,
 			IssuerImpl: &fake.Issuer{
 				FakeSign: func(context.Context, *cmapi.CertificateRequest) (*issuer.IssueResponse, error) {
 					return nil, errors.New("unexpected sign call")
 				},
 			},
 			Builder: &testpkg.Builder{
-				CertManagerObjects: []runtime.Object{gen.CertificateRequest("test")},
-				ExpectedActions:    []testpkg.Action{}, // no update
+				CertManagerObjects: []runtime.Object{gen.CertificateRequest("test"),
+					gen.Issuer("ca-issuer",
+						gen.AddIssuerCondition(cmapi.IssuerCondition{
+							Type:   cmapi.IssuerConditionReady,
+							Status: cmapi.ConditionTrue,
+						}),
+						gen.SetIssuerCA(cmapi.CAIssuer{}),
```

Comment:
It looks like the issuer has changed from being a selfsigned issuer to a CA issuer. Also, this CAIssuer doesn't specify a SecretName. Is that intentional?